### ISSUE TITLE
Don't upscale still frames in the video

### ISSUE THREAD

**Issue by Sveagruva** [hollowaykeanho/Upscaler#98]
### Description

Don't upscale frames that don't change from a previous frame. It's useful for upscaling anime where video has a lot of repeating frames

### Expected Behavior

Doesn't reupscale repeating frames

### Current Behavior

upscales all frames

**Comment by hollowaykeanho** [comment]
Mark this as help wanted because this requires some kind of on-going visual analytic algorithm.

What's the problem aside from taking a long time?

**Comment by Sveagruva** [comment]
No problems aside from redundant work being done. I don't think it requires complicated algorithm. When upscaling a frame check if it matches previous one (pixel by pixel) and if it does just copy previously upscaled frame instead of upscaling it.

**Comment by hollowaykeanho** [comment]
Yeah.. that needs the image recognition algorithm to analyze pixel differences. I'll see what I can do. Mark this as optimization feature.
